lib/iris/tests/test_nimrod.py

```diff
@@ -37,6 +38,59 @@ def test_multi_field_load(self):
         )
         self.assertCML(cube, ("nimrod", "load_2flds.cml"))
 
+    @tests.skip_data
+    def test_huge_field_load(self):
+        # load a wide range of cubes with all meta-data variations
+        for datafile in {
+            "u1096_ng_ek07_precip0540_accum180_18km",
+            "u1096_ng_ek00_cloud3d0060_2km",
+            "u1096_ng_ek00_cloud_2km",
+            "u1096_ng_ek00_convection_2km",
+            "u1096_ng_ek00_convwind_2km",
+            "u1096_ng_ek00_frzlev_2km",
+            "u1096_ng_ek00_height_2km",
+            "u1096_ng_ek00_precip_2km",
+            "u1096_ng_ek00_precipaccum_2km",
+            "u1096_ng_ek00_preciptype_2km",
+            "u1096_ng_ek00_pressure_2km",
+            "u1096_ng_ek00_radiation_2km",
+            "u1096_ng_ek00_radiationuv_2km",
+            "u1096_ng_ek00_refl_2km",
+            "u1096_ng_ek00_relhumidity3d0060_2km",
+            "u1096_ng_ek00_relhumidity_2km",
+            "u1096_ng_ek00_snow_2km",
+            "u1096_ng_ek00_soil3d0060_2km",
+            "u1096_ng_ek00_soil_2km",
+            "u1096_ng_ek00_temperature_2km",
+            "u1096_ng_ek00_visibility_2km",
+            "u1096_ng_ek00_wind_2km",
+            "u1096_ng_ek00_winduv3d0015_2km",
+            "u1096_ng_ek00_winduv_2km",
+            "u1096_ng_ek01_cape_2km",
+            "u1096_ng_umqv_fog_2km",
+            "u1096_ng_bmr04_precip_2km",
+            "u1096_ng_bsr05_precip_accum60_2km",
+            "probability_fields",
+        }:
+            cube = iris.load(
+                tests.get_data_path(("NIMROD", "uk2km", "cutouts", datafile,))
+            )
+            self.assertCML(cube, ("nimrod", f"{datafile}.cml"))
+
+    @tests.skip_data
+    def test_load_kwarg(self):
+        """Tests that the handle_metadata_errors kwarg is effective by setting it to
+        False with a file with known incomplete meta-data (missing ellipsoid)."""
+        datafile = "u1096_ng_ek00_pressure_2km"
+        with self.assertRaisesRegex(
+            TranslationError,
+            "Ellipsoid not supported, proj_biaxial_ellipsoid:-32767, horizontal_grid_type:0",
+        ):
+            iris.load(
+                tests.get_data_path(("NIMROD", "uk2km", "cutouts", datafile,)),
+                handle_metadata_errors=False,
```

Comment:
For the load api extension proposed in #3720, this isn't the correct syntax.
Instead, it will need to set the `loader_kwargs` keyword,
e.g. in this case ..., loader_kwargs={'handle_metadata_errors':False})

Reply:
Of, course, #3720 isn't agreed yet,either ...